### ISSUE TITLE
Possibility to use Early Stopping criteria when fine-tuning

### ISSUE THREAD

**Issue by Halli20** [google-research/text-to-text-transfer-transformer#187]
Hey I'm using the T5-3B Model in Google Colab and was wondering if it is possible to implement early stopping criteria when fine-tuning the model in order to limit the number of steps. 
The perfect solution would be to implement early stopping according to the validation loss. 
Is there a possibility to do this? And if not (with the validation loss )is it possible to do that by using the training loss?

**Comment by sharannarang** [comment]
We evaluate the validation metrics on checkpoints as the model is training and pick the one that has the best validation score to finally evaluate on the test set. So, we "early stop" the model by simply picking the appropriate checkpoint. There are no code changes needed for this. Hope this helps.
